samcli/local/lambda_service/local_lambda_invoke_service.py

```diff
@@ -46,14 +48,65 @@ def create(self):
                                methods=['POST'],
                                provide_automatic_options=False)
 
+        # setup request validation before Flask calls the view_func
+        self._app.before_request = self.validate_request
+
         self._construct_error_handling()
 
+    def validate_request(self):
+        """
+        Validates the incoming request
+
+        Returns
+        -------
+        flask.Response
+            A Response is returned in the following cases:
+            1. The Request data is not json serializable
+            2. Query Parameters are sent to the endpoint
+            3. The Request Content-Type is not application/json
+
+            Otherwise this function does not return anything
```

Comment:
I can imagine the case where we update the code but not comment. So this strong statement might not be true. I'd recommend removing the bullet list altogether or removing the assertion

Reply:
While your point is valid, that is an excuse I see frequently on why comments should never exist and really don't buy it. If we can't keep the code and comments updated together than we have bigger problems (imo). Unless there is another reason, I am going to leave this as is.